src/VisualStudio/Core/Def/Implementation/Diagnostics/VisualStudioWorkspaceDiagnosticAnalyzerProviderService.cs

```diff
@@ -29,16 +31,39 @@ public VisualStudioWorkspaceDiagnosticAnalyzerProviderService(VisualStudioWorksp
             // Get the analyzer assets for installed VSIX extensions through the VSIX extension manager.
             var extensionManager = workspace.GetVsService<SVsExtensionManager, IVsExtensionManager>();
 
-            var builder = ImmutableArray.CreateBuilder<HostDiagnosticAnalyzerPackage>();
-            foreach (var extension in extensionManager.GetEnabledExtensions(AnalyzerContentTypeName))
+            // get rootfolder and shellfolder location
+            string rootFolder;
+            string shellFolder;
+            if (TryGetRootAndShellFolder(extensionManager, out shellFolder, out rootFolder))
```

Comment:
@mavasani wouldn't that be equally scary if not more? this at least does exactly same as Extension Manager. hopefully, we get actual API from them so that we don't need to do this. but until then, I think this is less scary than having our own mechanism.